src/components/Select.tsx

```diff
@@ -0,0 +1,70 @@
+import { Fragment } from 'react';
+import { Listbox, Transition } from '@headlessui/react';
+import { CheckIcon, ChevronUpIcon, ChevronDownIcon } from '@heroicons/react/outline';
+import classNames from 'classnames';
+
+export interface SelectOption<V> {
+  label: string;
+  value: V;
+}
+
+export interface SelectProps<V> {
+  value: SelectOption<V>;
+  onChange: (value: SelectOption<V>) => void;
+  labelKey?: string;
+  options: SelectOption<V>[];
+}
+
+export function Select<V>({ value, onChange, options }: SelectProps<V>) {
+  const selected = value.label;
+
+  return (
+    <Listbox value={value} onChange={onChange}>
+      {({ open }) => (
+        <div className="relative w-full">
+          <Listbox.Button
+            className={classNames(
+              'relative flex w-full cursor-pointer flex-row items-center justify-between rounded-lg border-2 border-solid border-gray-800 bg-transparent p-2 text-left placeholder-gray-500 shadow-md hover:border-white  focus:placeholder-transparent focus:shadow-none focus:outline-none focus:ring-0',
```

Comment:
It would be awesome to break these long class names into more digestible units, taking advantage of `clsx`. I like breaking down more or less by state on the outermost, and by "thing being styled" in inner units, e.g.

```suggestion
              [
                'relative flex w-full', 'flex-row items-center justify-between p-2 text-left', 
                'rounded-lg border border-2 border-solid border-gray-800 bg-transparent shadow-md placeholder-gray-500', 'cursor-pointer'
              ], 
              [
                'hover:border-white  focus:placeholder-transparent focus:shadow-none focus:outline-none focus:ring-0'
              ]
```

Reply:
Why the [ ] @austinmilt ? Could it not just be separate regular arguments? :) 

```
'relative flex w-full', 'flex-row items-center justify-between p-2 text-left', 
'rounded-lg border border-2 border-solid border-gray-800 bg-transparent shadow-md placeholder-gray-500', 'cursor-pointer',
'hover:border-white  focus:placeholder-transparent focus:shadow-none focus:outline-none focus:ring-0'
```

Reply:
That being said I think it's more or less fine for this className to be long as it is basically ripped from the styled Tailwind example and well encapsulated into the low level react component itself.

Reply:
The square brackets, like the rest, are just in the name of readability. It's really hard (for me) to read a long class name and get an idea what the element will look like. It's a lot easier for me to read it in units like this:

`'relative flex w-full'`: how is this element positioned in its context
``flex-row items-center justify-between p-2 text-left'`: how does this element position its children
...
`'hover:border-white  focus:placeholder-transparent...'`: how does this element change when in a different state (e.g. hover)

I like to put the square brackets around to separate major categories. Everything in the default state would be wrapped in an outer brackets, and other states in another set of brackets, like `['default state stuff'], ['large screen stuff'], ['hover/focus stuff']`. Since each "state" might have multiple class name strings broken down by, e.g. self position vs child position, I like to group them in the square brackets for that state.

Obviously it's not a strict rule, but it helps me when class names are otherwise very long.

Reply:
> That being said I think it's more or less fine for this className to be long as it is basically ripped from the styled Tailwind example and well encapsulated into the low level react component itself.

I dont see how this makes a difference. As soon as the code was copy/pasted it became our code, so it should follow our rules

Reply:
Its overkill to use clsx on every class declaration in order to break up classnames. If folks want to follow a pattern I recommend following a loose ordering where it goes:
- positioning (eg flex, text align, grid)
- box sizing (eg padding, margin)
- styling (eg text color, bg color, border size)
- active states (eg transitions, hover, focus)

When clsx is used for conditionally adding styles I will look to break them up per line. I don't see much benefit from using array notation vs declaring one item per line. This is what I've done here.